### ISSUE TITLE
PHP warnings are terminating execution of the API

### ISSUE THREAD

**Issue by twinklebob** [Sebiworld/AppApi#24]
Warnings, which are usually seen as non fatal, are triggering the error reporting/handling function in the AppApi module. This stops execution even if, actually, they would've been fine.

In my case, I was getting this issue on a deployment server and was able to replicate it locally by changing the `disable_functions` setting in `php.ini` to include `set_time_limit` and then try and send an email using WireMailSmtp in the API. This results in an error similar to:

```json
{
    "error": "Internal Server Error",
    "devmessage": {
        "message": "set_time_limit() has been disabled for security reasons",
        "location": "/home/runcloud/webapps/D2U-API/site/assets/cache/FileCompiler/site/modules/WireMailSmtp-master/WireMailSmtp.module",
        "line": 748
    }
}
```

**Comment by twinklebob** [comment]
We've been able to enable that function on our server, which solves that issue for now, but there is a still a problem in the way warnings are handled.

It might be useful to get the warning output as part of the final API output, since there isn't really any other way to spot them, unless you go looking for the warning, but it mustn't stop execution.

I've got a PR that at least covers part of this.

**Comment by Sebiworld** [comment]
Hey @twinklebob , I've just committed some things that should be published as v1.1.6 soon. 

Your problem that the PHP warnings are caught too early by the module handler is not automatically solved by this. But I have added the new hook-enabled function `$this->registerErrorHandlers`, which you should be able to override with your own hook. You could then insert the handlers that you implemented in your pull request. 

In general, I don't want to change the warning handler yet, as long as I'm not sure that PHP doesn't echo something before the JSON output. Because that would make the complete output unusable...

I hope that you can solve the problem for the time being without having to adapt the module code. Please let me know briefly if it worked. Then I will publish the new version.

**Comment by Sebiworld** [comment]
v1.1.6 is out now. I hope this enables you to handle the warnings yourself? I close this issue for now...
